feature_engine/timeseries/forecasting/lag_features.py

```diff
@@ -0,0 +1,313 @@
+        check_is_fitted(self)
+
+        # check if 'X' is a dataframe
+        X = _is_dataframe(X)
+
+        # Check if input data contains same number of columns as dataframe used to fit.
+        _check_input_matches_training_df(X, self.n_features_in_)
+
+        # check if dataset contains na
+        if self.missing_values == "raise":
+            _check_contains_na(X, self.variables_)
+            _check_contains_inf(X, self.variables_)
+
+        # if freq is not None, it overrides periods.
+        if self.freq is not None:
+
+            if isinstance(self.freq, list):
+                df_ls = []
+                for fr in self.freq:
+                    tmp = X[self.variables_].shift(
+                        freq=fr,
+                        axis=0,
+                    )
+                    df_ls.append(tmp)
+                tmp = pd.concat(df_ls, axis=1)
+
+            else:
+                tmp = X[self.variables_].shift(
+                    freq=self.freq,
+                    axis=0,
+                )
+
+        else:
+            if isinstance(self.periods, list):
+                df_ls = []
+                for pr in self.periods:
+                    tmp = X[self.variables_].shift(
+                        periods=pr,
+                        axis=0,
+                    )
+                    df_ls.append(tmp)
+                tmp = pd.concat(df_ls, axis=1)
+
+            else:
+                tmp = X[self.variables_].shift(
+                    periods=self.periods,
+                    axis=0,
+                )
+
+        tmp.columns = self.get_feature_names_out(self.variables_)
+
+        X = X.merge(tmp, left_index=True, right_index=True, how="left")
+
+        if self.drop_original:
+            X = X.drop(self.variables_, axis=1)
+
+        return X
+
+    def get_feature_names_out(self, input_features: Optional[List] = None) -> List:
+        """Get output feature names for transformation.
+
+        Parameters
+        ----------
+        input_features: list, default=None
+            Input features. If `input_features` is `None`, then the names of all the
+            variables in the transformed dataset (original + new variables) is returned.
+            Alternatively, only the names for the lag features derived from
+            input_features will be returned.
+
+        Returns
+        -------
+        feature_names_out: list
+            The feature names.
+        """
+        check_is_fitted(self)
+
+        # Create names for all lag features or just the indicated ones.
+        if input_features is None:
+            # Create all lag features.
+            input_features_ = self.variables_
+        else:
+            if not isinstance(input_features, list):
+                raise ValueError(
+                    f"input_features must be a list. Got {input_features} instead."
+                )
+            # Create just indicated lag features.
+            input_features_ = input_features
```

Comment:
Add a control to check that the user entered variable belongs to self.variables_

Otherwise, this function will output names for lag features that were not created. For example if user passes a string with categorical variables.

Reply:
@Morgan-Sell  I discovered a bug here, so I will probably make another PR to your repo tomorrow :/